From 40c9bb2151081e6dcf06011808f389f044ff5e95 Mon Sep 17 00:00:00 2001
From: Tam Nguyen <tam.nguyen.xa@renesas.com>
Date: Wed, 28 May 2025 02:53:09 +0700
Subject: [PATCH 38/46] arm64: defconfig: Enable Ethernet 1G support for R-Car
 X5H

Enable relevant configuration options to support Ethernet 1G on the
Renesas R-Car X5H (r8a7800) platform, codenamed Ironhide as well as
update the size of secure register area.

Connection topology:
  X5H/RSwitch3 <--> X5H/Ethernet PCS <--> X5H/MP-PHY <--> Ironhide/TI PHY

This enables proper initialization of the Ethernet pipeline from MAC
to PHY through the PCS and MP-PHY layers.

Signed-off-by: Tam Nguyen <tam.nguyen.xa@renesas.com>
Signed-off-by: Thanh Quan <thanh.quan.xn@renesas.com>
---
 arch/arm64/boot/dts/renesas/r8a78000.dtsi | 2 +-
 arch/arm64/configs/defconfig              | 3 +++
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/renesas/r8a78000.dtsi b/arch/arm64/boot/dts/renesas/r8a78000.dtsi
index a5c0c3bdc982..a8633ef436ba 100644
--- a/arch/arm64/boot/dts/renesas/r8a78000.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a78000.dtsi
@@ -943,7 +943,7 @@ mp_phy: mp_phy@c9a00000 {
 		rswitch3: ethernet@c9bc0000 {
 			compatible = "renesas,r8a78000-ether-switch";
 			reg = <0 0xc9bc0000 0 0x0040000>,  // base
-				  <0 0xc9b80000 0 0x00240000>; // secure_base
+				  <0 0xc9b80000 0 0x00040000>; // secure_base
 			reg-names = "base", "secure_base";
 			interrupts = <GIC_SPI 799 IRQ_TYPE_LEVEL_HIGH>,
 				 <GIC_SPI 800 IRQ_TYPE_LEVEL_HIGH>,
diff --git a/arch/arm64/configs/defconfig b/arch/arm64/configs/defconfig
index 9f9eb8edf133..f323f7d4e00e 100644
--- a/arch/arm64/configs/defconfig
+++ b/arch/arm64/configs/defconfig
@@ -396,6 +396,7 @@ CONFIG_AT803X_PHY=y
 CONFIG_REALTEK_PHY=y
 CONFIG_ROCKCHIP_PHY=y
 CONFIG_DP83867_PHY=y
+CONFIG_DP83869_PHY=y
 CONFIG_VITESSE_PHY=y
 CONFIG_MDIO_BUS_MUX_MULTIPLEXER=y
 CONFIG_MDIO_BUS_MUX_MMIOREG=y
@@ -1287,6 +1288,8 @@ CONFIG_PHY_QCOM_USB_SNPS_FEMTO_V2=m
 CONFIG_PHY_QCOM_USB_HS_28NM=m
 CONFIG_PHY_QCOM_USB_SS=m
 CONFIG_PHY_R8A779F0_ETHERNET_SERDES=y
+CONFIG_PHY_R8A78000_ETHERNET_PCS=y
+CONFIG_PHY_R8A78000_MP_PHY=y
 CONFIG_PHY_RCAR_GEN3_PCIE=y
 CONFIG_PHY_RCAR_GEN3_USB2=y
 CONFIG_PHY_RCAR_GEN3_USB3=m
-- 
2.34.1

